crates/task-impls/src/transactions.rs

```diff
@@ -342,74 +342,85 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> TransactionTaskState<TYPES,
+
+                // First we complete the query to the fastest fraction of the builders
+                let threshold = (self.builder_clients_v0_1.len() * BUILDER_MAIN_BATCH_THRESHOLD_DIVIDEND)
+                    .div_ceil(BUILDER_MAIN_BATCH_THRESHOLD_DIVISOR);
+                let mut tasks = tasks.take(threshold);
+                while let Some(result) = tasks.next().await {
+                    results.push(result);
+                    if query_start.elapsed() > BUILDER_MAIN_BATCH_CUTOFF {
+                        break;
+                    }
+                }
 
-        // Then we query the rest, alotting additional `elapsed * BUILDER_ADDITIONAL_TIME_MULTIPLIER`
-        // for them to respond. There's a fixed floor of `BUILDER_MINIMUM_QUERY_TIME` for both
-        // phases
-        let timeout = async_sleep(std::cmp::max(
-            query_start
-                .elapsed()
-                .mul_f32(BUILDER_ADDITIONAL_TIME_MULTIPLIER),
-            BUILDER_MINIMUM_QUERY_TIME.saturating_sub(query_start.elapsed()),
-        ));
-        futures::pin_mut!(timeout); // Stream::next requires Self::Unpin
-        let mut tasks = tasks.into_inner().take_until(timeout);
-        while let Some(result) = tasks.next().await {
-            results.push(result);
+                // Then we query the rest, alotting additional `elapsed * BUILDER_ADDITIONAL_TIME_MULTIPLIER`
+                // for them to respond. There's a fixed floor of `BUILDER_MINIMUM_QUERY_TIME` for both
+                // phases
+                let timeout = async_sleep(std::cmp::max(
+                    query_start
+                        .elapsed()
+                        .mul_f32(BUILDER_ADDITIONAL_TIME_MULTIPLIER),
+                    BUILDER_MINIMUM_QUERY_TIME.saturating_sub(query_start.elapsed()),
+                ));
+                futures::pin_mut!(timeout); // Stream::next requires Self::Unpin
+                let mut tasks = tasks.into_inner().take_until(timeout);
+                while let Some(result) = tasks.next().await {
+                    results.push(result);
+                }
+
+                results
+                    .into_iter()
+                    .filter_map(|result| match result {
+                        Ok(value) => Some(value),
+                        Err(err) => {
+                            tracing::warn!(%err, "Error getting available blocks");
+                            None
+                        }
+                    })
+                    .flatten()
+                    .collect::<Vec<_>>()
+            }}
         }
 
-        results
-            .into_iter()
-            .filter_map(|result| match result {
-                Ok(value) => Some(value),
-                Err(err) => {
-                    tracing::warn!(%err, "Error getting available blocks");
-                    None
-                }
-            })
-            .flatten()
-            .collect::<Vec<_>>()
+        if version.minor == 3 {
```

Comment:
You might already be planning to do this, but I think this should be an inequality, and not just on the minor version. e.g. `if version >= MARKETPLACE_VERSION`.

I have some thoughts about setting version numbers that might be worth discussing; I'll share them on Zulip